### DISCUSSION TITLE
A read problem

### DISCUSSION THREAD

**Discussion by fastkits** [networkx/networkx#4854] [Q&A] [answered]
G = nx.MultiGraph()
    G.add_edge(1, 3, data={'s': 1})
    print(G.nodes)  # -----> [1, 3]
    nx.write_gml(G, "test.gml")
    G = nx.read_gml("test.gml")
    print(G.nodes)  # -----> ['1', '3']

    # [1, 3]  ---------->  ['1', '3']  

    Why not? What's the solution

**Comment by dschult** [comment, marked as answer]
It seems that you need to specify a `destringizer` argument to read_gml. 
G = nx.rad_gml("test.gml", destringizer=int)
seems to at least get rid of this specific trouble.

**Reply by rossbar** [reply]
Thanks Dan --- I think this answers the specific question quite thoroughly!
